src/lib/InstantSearch.ts

```diff
@@ -234,33 +223,29 @@ See ${createDocumentationLink({
       this._searchFunction = searchFunction;
     }
 
-    const defaultRoutingOptions = {
-      stateMapping: simpleStateMapping(),
-      router: historyRouter(),
-    };
-
-    let routingOptions: Routing | null = null;
-    if (routing === true) {
-      routingOptions = defaultRoutingOptions;
-    } else if (isPlainObject(routing)) {
-      routingOptions = {
-        ...defaultRoutingOptions,
-        ...routing,
-      };
+    if (routing) {
+      const routerOptions = typeof routing === 'boolean' ? undefined : routing;
+      this.use(createRouter(routerOptions));
     }
+  }
 
-    if (routingOptions) {
-      this._routingManager = new RoutingManager({
-        ...routingOptions,
-        instantSearchInstance: this,
-      });
+  public use(...middleware: Middleware[]): this {
+    const newMiddlewareList = middleware.map(fn => {
+      const newMiddleware = fn({ instantSearchInstance: this });
+      this.middleware.push(newMiddleware);
```

Comment:
```suggestion
      this.middlewares.push(newMiddleware);
```

(also needs to be updated in more places if this is chosen

Reply:
"Middlewares" is not correct english though.